kiosk/src/Components/AddingGame.tsx

```diff
@@ -0,0 +1,142 @@
+import { useEffect, useState } from "react";
+import { Kiosk } from "../Models/Kiosk";
+import { KioskState } from "../Models/KioskState";
+import configData from "../config.json"
+import "../Kiosk.css";
+import AddGameButton from "./AddGameButton";
+import {QRCodeSVG} from 'qrcode.react';
+import { generateKioskCodeAsync, getGameCodeAsync, isLocal } from "../BackendRequests";
+interface IProps {
+    kiosk: Kiosk
+  }
+
+const AddingGame: React.FC<IProps> = ({ kiosk }) => {
+    // TODO: update the urls to be more flexible for production code.
+    const [kioskCode, setKioskCode] = useState("");
+    const [renderQRCode, setRenderQRCode] = useState(true);
+    const [menuButtonSelected, setMenuButtonState] = useState(false);
+    const [qrCodeButtonSelected, setQrButtonState] = useState(false);
+    const kioskCodeUrl = isLocal() ? "http://localhost:3000/static/kiosk/" : "/kiosk";
+
+    const updateLoop = () => {
+        if (!menuButtonSelected && kiosk.gamepadManager.isDownPressed()) {
+            setMenuButtonState(true);
+            if (qrCodeButtonSelected) {
+                setQrButtonState(false);
+            }
+        }
+        if (menuButtonSelected && kiosk.gamepadManager.isAButtonPressed()) {
+            kiosk.showMainMenu();
+        }
+        if (!renderQRCode && kiosk.gamepadManager.isUpPressed()) {
+            setMenuButtonState(false);
+            setQrButtonState(true);
+        }
+        if (qrCodeButtonSelected && kiosk.gamepadManager.isAButtonPressed()) {
+            setRenderQRCode(true);
+        }
+    }
+
+    useEffect(() => {
+        let intervalId: any = null;
+        intervalId = setInterval(() => {
+            updateLoop();
+        }, configData.GamepadPollLoopMilli);
+        
+        return () => {
+            if (intervalId) {
+                clearInterval(intervalId);
+            }
+        };
+
+    });
+
+    useEffect(() => {
+        async function generateKioskCode() {
+            try {
+                const newKioskCode: string = await generateKioskCodeAsync();
+                setKioskCode(newKioskCode);
+                await addGameToKiosk(newKioskCode);
+            }
+            catch (error) {
+                setRenderQRCode(false);
+                throw new Error("Unable to generate kiosk code");
+            }
+        }
+
+        generateKioskCode();
+
+        function addGameToKiosk(kioskCode: string) {
+            const timeoutDuration = 600000; // wait for 10 minutes until the kiosk code expires
+            const whenToPoll = 5000; // wait for five seconds to poll for game code data
+            if (kiosk.state !== KioskState.AddingGame) {
+                return;
+            }
+            return new Promise<void>((resolve, reject) => {
+                let pollInterval: any;
+                let pollTimeout: any;
+                pollInterval = setInterval(async () => {
```

Comment:
if the backend is slow to respond for some reason (like bad internet), this has the potential to let requests build up a bit. maybe add a bit of logic to make sure we wait for the current request to complete before making another one.

Reply:
I will do this as an update tomorrow.